### PR TITLE
Removed minting timelock and minimum time between mints, and modified test cases

### PR DESCRIPTION
**Pull request by xenide**

### CODE REVIEW

File: contracts/Uni.sol [hunk 1]
```diff
@@ -19,13 +19,8 @@ contract Uni {
...
 
-    /// @notice The timestamp after which minting may occur
-    uint public mintingAllowedAfter;
-
-    /// @notice Minimum time between mints
-    uint32 public constant minimumTimeBetweenMints = 1 days * 365;
-
     /// @notice Cap on the percentage of totalSupply that can be minted at each mint
+    /// TODO: Need to think and decide on whether to keep this or not  
     uint8 public constant mintCap = 2;
```
Comment (xenide) [inline contracts/Uni.sol]: We need to decide if we still want this

Comment (OliverNChalk) [reply contracts/Uni.sol]: We should remove the mintCap from core contract, this can be introduced through a `minter` contract.

Comment (xenide) [reply contracts/Uni.sol]: Okay so should we write a `minter` contract at this stage? What other functionalities do we need in this `minter`?

Comment (xenide) [reply contracts/Uni.sol]: Also, done removing the mintCap

**Comment by OliverNChalk** [comment]
CI  is complaining about our use of the experimental ABIEncoder? Do we know why Uniswap used that flag?

**Comment by xenide** [comment]
> CI is complaining about our use of the experimental ABIEncoder? Do we know why Uniswap used that flag?

Seems that the `pragma experimental` statement is needed for the function to take in string[] and bytes[] as arguments. Without it, compilation fails. Also, it seems that only `GovernorAlpha.sol` is using these structures, so the statement in `Uni.sol` is useless, we can remove that. 

If we really want to avoid this, we have to upgrade to solidity ^0.8.0, which natively supports those structures

**Comment by OliverNChalk** [comment]
Hmm, Uniswap's later governance deployments keep `pragma experimental`. I think we should just leave it for now because if it's safe enough for them it should be good enough for us. We can always migrate to a newer version later.

**Review by OliverNChalk — APPROVED** [review]

**Pull request merged**